src/hdmf/data_utils.py

```diff
@@ -20,7 +20,10 @@ def append_data(data, arg):
         data.append(arg)
         return data
     elif isinstance(data, np.ndarray):
-        return np.append(data,  np.expand_dims(arg, axis=0), axis=0)
+        if isinstance(arg, np.ndarray):
+            return np.append(data, arg)
```

Comment:
I tried this:
```python
a = np.array([[1, 2, 3], [4, 5, 6]])
b = np.array([7, 8, 9])
np.append(a, b)
```
The result is that the array is flattened: `array([1, 2, 3, 4, 5, 6, 7, 8, 9])`

That's not good and could break things.

I get that you want to use `append` for structured arrays. Can you change the `if` to check that it is a structured array instead of any numpy array?

Reply:
I think this issue returns with your latest commit... please review and add a test for this case.